pyvista/core/filters/poly_data.py

```diff
@@ -550,9 +550,19 @@ def merge(
                 polydata_merged = pyvista.PolyData(
                     merged.points, faces=merged.cells, n_faces=merged.n_cells, deep=False
                 )
+                # Calling update() will modify the active scalars in this specific
+                # case. Store values to restore after updating.
```

Comment:
I checked if it is possible to handle this in the update method, but it could not be handled when main_has_priority=True. (This is a memo.)